controllers/complianceeventsapi/migrations/000001_compliance_history_initial_tables.up.sql

```diff
@@ -0,0 +1,57 @@
+BEGIN;
+
+CREATE TABLE IF NOT EXISTS clusters(
+   id serial PRIMARY KEY,
+   name TEXT NOT NULL,
+   cluster_id TEXT UNIQUE NOT NULL,
+   UNIQUE (name, cluster_id)
+);
+
+CREATE INDEX IF NOT EXISTS idx_clusters_name ON clusters (name);
+
+CREATE TABLE IF NOT EXISTS parent_policies(
+   id serial PRIMARY KEY,
+   name TEXT NOT NULL,
+   categories TEXT [],
+   controls TEXT [],
+   standards TEXT [],
+   UNIQUE (name, categories, controls, standards)
+);
+
+CREATE TABLE IF NOT EXISTS policies(
+   id serial PRIMARY KEY,
+   kind TEXT NOT NULL,
+   api_group TEXT NOT NULL,
+   name TEXT NOT NULL,
+   parent_policy_id INT NOT NULL,
+   spec TEXT,
+   -- SHA1 hash
+   spec_hash CHAR[40],
+   CONSTRAINT fk_parent_policy_id
+      FOREIGN KEY(parent_policy_id) 
+	  REFERENCES parent_policies(id),
+   UNIQUE (kind, api_group, name, parent_policy_id, spec_hash)
+);
+
+CREATE INDEX IF NOT EXISTS idx_policies_spec_hash ON policies (spec_hash);
+
+CREATE TABLE IF NOT EXISTS compliance_events(
+   id serial PRIMARY KEY,
+   cluster_id INT NOT NULL,
+   policy_id INT NOT NULL,
+   compliance TEXT NOT NULL,
+   message TEXT NOT NULL,
+   timestamp TIMESTAMP NOT NULL,
+   metadata TEXT,
+   CONSTRAINT fk_policy_id
+      FOREIGN KEY(policy_id) 
+	  REFERENCES policies(id),
+   CONSTRAINT fk_cluster_id
+      FOREIGN KEY(cluster_id) 
+	  REFERENCES clusters(id)
+);
+
+CREATE INDEX IF NOT EXISTS idx_compliance_events_compliance ON compliance_events (compliance);
+CREATE INDEX IF NOT EXISTS idx_compliance_events_timestamp ON compliance_events (timestamp);
+
+COMMIT;
```

Comment:
Some other questions:
- how do I tell if a policy was deleted or disabled from the data above?
- can I tell if the policy stayed the same but the source changed?
- should we log the policy remediation so changes there are reflected?
- I expect we could expand to include PolicyAutomation that is triggered in the future -- plus any other integrations we may support in the future...
- I think it's good for now to not focus much on policy placement for this historical data, but I wonder just like with the policy being deleted or disabled if we should integrate in "some" form of placement history.  With placement becoming more complex, this can be discussed later.

Reply:
1. I added this to the design in this [PR|https://github.com/open-cluster-management-io/enhancements/pull/101]. It'll be a separate compliance event from the Spec Sync controller.
2. Could you please rephrase this question?
3. Perhaps, but at the moment I'm thinking not for PolicyAutomation because we are recording the policy state and if the PolicyAutomation changes this state, the compliance event will be recorded through the lens of the policy.
4. My solution for the disabled case is the Spec Sync controller would send an additional compliance event where the `compliance` is set to `deleted`. This does not track the placement but tracks the actual state on the cluster which is better because a cluster could not receive its placement update until much later if the cluster is disconnected from the hub. Having this separate compliance in the `compliance_events` table can give you a sense of when a policy was deleted and then reapplied.

Reply:
I don't remember what I was thinking with the comment:
- can I tell if the policy stayed the same but the source changed?

Regarding remediation:  
- should we log the policy remediation so changes there are reflected? I think I'm ok ignoring this comment for now too.